clients/da-vinci-client/src/main/java/com/linkedin/davinci/kafka/consumer/LeaderFollowerStoreIngestionTask.java

```diff
@@ -2991,8 +2992,8 @@ protected void processMessageAndMaybeProduceToKafka(
         if (isWriteComputationEnabled && partitionConsumptionState.isEndOfPushReceived()) {
           partitionConsumptionState.setTransientRecord(kafkaClusterId, consumerRecord.getOffset(), keyBytes, -1, null);
         }
-        leaderProducedRecordContext =
-            LeaderProducedRecordContext.newDeleteRecord(kafkaClusterId, consumerRecord.getOffset(), keyBytes, null);
+        leaderProducedRecordContext = LeaderProducedRecordContext
```

Comment:
nit: is it possible to write unit tests for catching such issues? Your integ test is great BTW

Reply:
Oops I missed this review. Yeah, definitely